src/cron/updateCurrentStreak.ts

```diff
@@ -0,0 +1,40 @@
+import { Cron } from './cron';
+import { User, UserStreak } from '../entity';
+import { checkAndClearUserStreak } from '../common';
+
+const cron: Cron = {
+  name: 'update-current-streak',
+  handler: async (con, logger) => {
+    try {
+      await con.transaction(async (entityManager): Promise<void> => {
+        const usersPastStreakTime = await entityManager
+          .createQueryBuilder()
+          .select(
+            `us.*, (date_trunc('day', us."lastViewAt" at time zone COALESCE(u.timezone, 'utc'))::date) AS "lastViewAtTz", u.timezone`,
+          )
+          .from(UserStreak, 'us')
+          .innerJoin(User, 'u', 'u.id = us."userId"')
+          .where(`us."currentStreak" != 0`)
+          .andWhere(
+            `(date_trunc('day', us. "lastViewAt" at time zone COALESCE(u.timezone, 'utc'))::date) < (date_trunc('day', now() at time zone COALESCE(u.timezone, 'utc'))::date) - interval '1 day' `,
+          )
+          .getRawMany();
+
+        await Promise.all(
+          usersPastStreakTime.map(async (userStreak) => {
+            return await checkAndClearUserStreak(
```

Comment:
This function runs already the query, which means if we have 10k users to reset the streak, we will have 10k queries. We can instead keep an array of user IDs to reset and then do it in a single query or batch

Reply:
Hmm yeah we can extract to a `checkUserStreak` and seperate query for clearing here?

Reply:
Yes, would be better to separate concerns